cupy/core/core.pyx

```diff
@@ -2350,7 +2350,7 @@ cpdef ndarray _repeat(ndarray a, repeats, axis=None):
     # Scalar and size 1 'repeat' arrays broadcast to any shape, for all
     # other inputs the dimension must match exactly.
     cdef bint broadcast = False
-    if isinstance(repeats, int):
+    if numpy.issubdtype(repeats, numpy.integer):
```

Comment:
This was necessary because `cupy.pad` internally uses `cupy.repeat` and
the `repeats` argument can have numpy int types.